orchestra/models.py

```diff
@@ -28,16 +76,92 @@ class Certification(models.Model):
     name = models.CharField(max_length=200)
     description = models.TextField()
-    required_certifications = models.ManyToManyField('self',
-                                                     blank=True)
+    required_certifications = models.ManyToManyField(
+        'self',
+        blank=True,
+        related_name='dependent_certifications',
+        symmetrical=False)
+    workflow = models.ForeignKey(Workflow, related_name='certifications')
 
     def __str__(self):
         return '{}'.format(self.slug)
 
+    class Meta:
+        app_label = 'orchestra'
+        unique_together = (('workflow', 'slug'),)
+
+
+class WorkflowStep(models.Model):
+    """
+    WorkflowSteps represent individual tasks in a workflow version.
+
+    Attributes:
+        slug (str):
+            Unique identifier for the workflow step.
+        name (str):
+            Human-readable name for the workflow step.
+        description (str):
+            A longer description of the workflow step.
+        workflow_version (orchestra.models.WorkflowVersion):
+            The workflow version that this is a step of.
+        creation_depends_on ([orchestra.models.WorkflowStep]):
+            Workflow steps that must be complete to begin this step.
+        submission_depends_on ([orchestra.models.WorkflowStep]):
+            Workflow steps that must be complete to end this step.
+        is_machine (bool):
+            True if this step is performed by a machine, not a human.
+        execution_function (str):
+            A JSON blob containing the path to and name of a python method that
+            will execute this step (only valid for machine steps).
+        required_certifications ([orchestra.models.Certification]):
+            The certifications a worker must have in order to work on this step
+            (only valid for human steps).
+        assignment_policy (str):
+            A JSON blob used to decide which worker to assign to this step
+            (only valid for human steps).
+        review_policy (str):
+            A JSON blob used to decide whether or not to review this step
+            (only valid for human steps).
+        user_interface (str):
+            A JSON blob used to describe the files used in the user interface
+            for this step (only valid for human steps).
+    """
+    # General fields
+    slug = models.CharField(max_length=200)
+    name = models.CharField(max_length=200)
+    description = models.TextField()
+    workflow_version = models.ForeignKey(WorkflowVersion, related_name='steps')
+    creation_depends_on = models.ManyToManyField(
+        'self',
+        blank=True,
+        related_name='creation_dependents',
+        symmetrical=False)
+    submission_depends_on = models.ManyToManyField(
+        'self',
+        blank=True,
+        related_name='submission_dependents',
+        symmetrical=False)
+
+    # Machine step fields
+    is_machine = models.BooleanField()
```

Comment:
Hm, not sure I agree. Enums make code much more verbose (have to pass around `Step.WorkerType.HUMAN`s all over the place), and are unintuitive to serialize through JSON (i.e., our version.json manifests would have to either use the integer or some string name).

Can you think of a situation in the near-to-medium future where we have other types of workers? Right now, this field is mostly just sugar for detecting when a step has an execution_function vs. an interface.